components/net/connector.rs

```diff
@@ -50,9 +50,31 @@ impl NetworkConnector for HttpsConnector {
 
 pub type Connector = HttpsConnector;
 
-pub fn create_ssl_client(ca_file: &PathBuf) -> OpensslClient {
+pub fn create_ssl_client(certs: &str) -> OpensslClient {
+    let mut certs = certs.to_string();
+    // certs include multiple certificates. We could add all of them at once,
+    // but if any of them were already added, openssl would fail to insert all
+    // of them.
     let mut ssl_connector_builder = SslConnectorBuilder::new(SslMethod::tls()).unwrap();
-    ssl_connector_builder.set_ca_file(ca_file).expect("could not set CA file");
```

Comment:
Maybe we should add a function to the `openssl` crate that takes the certificate file contents as a string.  Then we could continue to let OpenSSL do the parsing here.

Reply:
It already has a method to parse the whole certs file, but it's a "all or non" operation (not like the previous method we were passing a file instead of a buffer). If it fails to load one of the certificate, they are all rejected. And in our case, it might fail because some certificates were already registered (so it's fine to ignore the error). So if we were to improve the Rust OpenSSL API, we should allow OpenSSL to ignore the duplicate error.

I can file a bug for that. Sounds to me that it's a OpenSSL issue, not a Rust OpenSSL problem though.